### PR TITLE
Added the Tiberian Sun harvest animation

### PR DESCRIPTION
**Pull request by Mailaender**
as discussed in https://github.com/OpenRA/OpenRA/pull/7194#discussion_r22291607 and IRC.

### CODE REVIEW

**Comment by pchote** [comment]
This:
screen shot 2015-01-02 at 18 10 58

is still quite different to the original:
screen shot 2015-01-02 at 18 12 37

**Comment by Mailaender** [comment]
Interesting. I though the green dust with yellow sparks made sense. Fixed the palette and tweaked the offset further.

**Comment by penev92** [comment]
Green does make more sense. Downside is when you are harvesting blue tiberium.

**Comment by Mailaender** [comment]
Rebased.

**Comment by penev92** [comment]
Do we have (can we make) a blue dust animation? I think it would be nice to have a colored animation depending on the tiberium instead of the grey one, although that won't be entirely historically accurate.

**Comment by Mailaender** [comment]
A dynamic palette modifier based on the terrain tile the harvester is sitting adds quite the complication to this. I am fine with the grey dust assuming we are not copying a Westwood oversight here.

**Comment by penev92** [comment]
Perhaps they also decided it was easier this way.
I'm fine with gray if dynamic color would be too complicated.

**Comment by penev92** [comment]
But why is the animation offset sideways? It doesn't look like it is in the original   :?

File: OpenRA.Mods.Common/Traits/Render/WithHarvestAnimation.cs [hunk 1]
```diff
@@ -65,5 +70,13 @@ public static int ZOffsetFromCenter(Actor self, WPos pos, int offset)
 			var delta = self.CenterPosition - pos;
 			return delta.Y + delta.Z + offset;
 		}
+
+		public IEnumerable<IRenderable> ModifyRender(Actor self, WorldRenderer wr, IEnumerable<IRenderable> r)
+		{
+			if (string.IsNullOrEmpty(info.Palette))
+				return r;
+			else
+				return r.Where(a => a.IsDecoration).Select(a => a.WithPalette(wr.Palette(info.Palette))).Concat(r.Where(a => !a.IsDecoration));
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithHarvestAnimation.cs]: This will replace the palette of all decorative animations, which is a giant hack.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithHarvestAnimation.cs]: Suggestions?

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithHarvestAnimation.cs]: Implement `IRender` and return the correctly-paletted renderables directly.

Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Render/WithHarvestAnimation.cs]: I found a way simpler solution. Updated.

**Comment by penev92** [comment]
Needs a rebase.
I still don't understand the animation's Y offset.

**Comment by Mailaender** [comment]
Rebased and removed the Y offset again.

**Comment by penev92** [comment]
:+1:

**Comment by abcdefg30** [comment]
:+1:

**Pull request merged**